### PR TITLE
Fix reclass config-file reading

### PR DESCRIPTION
**Pull request by MatteoVoges**
Fixes issue #969 

## Proposed Changes
- Refactor, fix and simplify the parsing / reading process of `reclass-config.yml` file
- It's unclear if the attribute `inventory_base_uri` is relative to its location (`inventory_path`) and if the subdirs `targets` (`nodes_uri`) and `classes` (`classes_uri`) should get their path from the `inventory_path` or the `inventory_base_uri`. I implemented it the way it was before ( --> `inventory_path`) but i would be nice to change that.

- We could add some documentation about this config file, especially with the new `compose-node-name` feature in #933 

--- 
### Contributed by  [](https://www.nexenio.com)

### CODE REVIEW

**Review by Moep90 — APPROVED** [review]

File: kapitan/resources.py [hunk 1]
```diff
@@ -315,50 +314,58 @@ def inventory_reclass(inventory_path, ignore_class_notfound=False):
...
-        except ReclassException as e:
-            if isinstance(e, NotFoundError):
-                logger.error("Inventory reclass error: inventory not found")
-            else:
-                logger.error("Inventory reclass error: %s", e.message)
-            raise InventoryError(e.message)
+    else:
+        logger.debug("Inventory reclass: No config file found. Using reclass inventory config defaults")
+
+    # normalise relative nodes_uri and classes_uri paths and up
```
Comment (MatteoVoges) [inline kapitan/resources.py:349]:
```suggestion
    # normalise relative nodes_uri and classes_uri paths
```

File: kapitan/resources.py [hunk 2]
```diff
@@ -315,50 +314,58 @@ def inventory_reclass(inventory_path, ignore_class_notfound=False):
...
-                reclass_config["storage_type"],
-                reclass_config["nodes_uri"],
-                reclass_config["classes_uri"],
-                reclass_config["compose_node_name"],
+    # get reclass config from file 'inventory/reclass-config.yml'
+    cfg_file = os.path.join(inventory_path, "reclass-config.yml")
+    if os.path.isfile(cfg_file):
+        with open(cfg_file, "r") as fp:
+            config = yaml.load(fp.read(), Loader=YamlLoader)
+            logger.debug("Using reclass inventory config at: %s", cfg_file)
```
Comment (MatteoVoges) [inline kapitan/resources.py:337]:
```suggestion
            logger.debug("Using reclass inventory config at: {}".format(cfg_file))
```

File: kapitan/resources.py [hunk 3]
```diff
@@ -315,50 +314,58 @@ def inventory_reclass(inventory_path, ignore_class_notfound=False):
...
+        if config:
+            # set attributes, take default values if not present
+            for key, value in config.items():
+                reclass_config[key] = value
+        else:
+            logger.debug(
+                "{}: Invalid yaml or empty. Using reclass inventory config defaults".format(cfg_file)
```
Comment (MatteoVoges) [inline kapitan/resources.py:344]:
```suggestion
                "{}: Empty config file. Using reclass inventory config defaults".format(cfg_file)
```

[the remark this answers is not in the document]
Comment (MatteoVoges) [reply kapitan/resources.py:344]: Youre right! I fixed it in the latest iteration

**Review by gburiola — APPROVED** [review]

**Comment by MatteoVoges** [comment]
@uberspot Thank you for merging. Could you check that you don't merge with the _rebase option_, because then several unfinished or WIP commits gets merged to master.

My personal preference would be using a squashed commit!

Thank you in advance!

**Comment by gburiola** [comment]
> @uberspot Thank you for merging. Could you check that you don't merge with the _rebase option_, because then several unfinished or WIP commits gets merged to master.
> 
> My personal preference would be using a squashed commit!
> 
> Thank you in advance!

I think that's a tricky one to get right in a repo with multiple people working. When I create PRs, specially when working on something a bit more complex, I try to separate the work into multiple commits with logical pieces of work that can be grouped together. If there's a need to fix, I personally rebase and delete commits when needed so that commit history is exactly how I want it to be.
For this reason, I do not like all of my commits squashed into one.

**Pull request merged**